src/jsMain/kotlin/dev.gitlive.difflib/unifieddiff/UnifiedDiffReader.kt

```diff
@@ -0,0 +1,18 @@
+package dev.gitlive.difflib.unifieddiff
+
+import kotlinx.coroutines.await
+import kotlin.js.Promise
+import kotlin.js.json
+
+val readline by lazy { js("require")("readline") }
+
+actual abstract class InputStream (val buffer: dynamic)
+
+actual suspend fun UnifiedDiffReader.Companion.readLine(stream: InputStream): UnifiedDiff {
+    val rl = readline.createInterface(json("input" to stream.buffer))
```

Comment:
if you called the prop on InputStream  "input" could you just pass stream in here directly instead of calling json()?